domain/domain.go

```diff
@@ -457,8 +457,8 @@ func (do *Domain) Close() {
 	if do.etcdClient != nil {
 		terror.Log(errors.Trace(do.etcdClient.Close()))
 	}
-	do.slowQuery.Close()
 	do.sysSessionPool.Close()
+	do.slowQuery.Close()
```

Comment:
Is it enough to put “do.slowQuery.Close()” after line462?

Reply:
`do.slowQuery.Close()` will make goroutine exit and WaitGroup count -1.
If this line is moved to line 462,`do.wg.Wait()` would wait forever.

Reply:
Ignore it.

Reply:
But I think it's useless. After this function is executed, "updateStatsWorker" may still be called by another goroutine. This problem still exists.
Maybe we can replace "ctx"(the updateStatsWorker's argument) with the session created in the sysSessionPool.

Reply:
Do you mean after session pool is closed, `updateStatsWorker`  will be still executing ? @zimulala

Reply:
Ye. Because the `session` used in statistic bootstrap is not get from `session pool`. @tiancaiamao

Reply:
Current session pool implementation will **block waiting resource to be put back**, so after the pool is closed, there won't be system session running. @crazycs520 @zimulala